src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,140 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] storage;
+    private int size;
+
+    public ArrayList() {
+        storage = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        storage = new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (isFilled()) {
+            increaseCapacity();
+        }
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (checkIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("index " + index
+                    + " is out of bound for length " + size);
+        }
+        if (isFilled()) {
+            increaseCapacity();
+        }
+        moveToRight(index);
+        storage[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (storage.length < size + list.size()) {
+            increaseCapacity(size + list.size());
+        }
+        for (int i = 0; i < list.size(); i++) {
+            storage[size++] = list.get(i);
+        }
     }
```

Comment:
useless check `storage.length < size + list.size()`, in this method you can add list elements via add method(which already has fill check) `storage[size++] = list.get(i)`  ->   `storage.add(list.get(i))`

Reply:
ok